### PR TITLE
Add quickstart iOS documentation

### PR DESCRIPTION
**Pull request by danielnugraha**
Adding new quickstart ios to the documentation page based on examples/ios. The quickstart page includes:

1. How to import flwr into xcode.
2. How to create flower ios client to federately train MNIST classifier model.
3. Showcasing the important steps to integrate FL using flower.

### CODE REVIEW

File: doc/source/index.rst [hunk 1]
```diff
@@ -53,6 +53,7 @@ A learning-oriented series of federated learning tutorials, the best place to st
    quickstart-mxnet
    quickstart-scikitlearn
    quickstart-xgboost
+   quickstart-ios
 
 QUICKSTART TUTORIALS: :ref:`PyTorch <quickstart-pytorch>` | :ref:`TensorFlow <quickstart-tensorflow>` | :ref:`🤗 Transformers <quickstart-huggingface>` | :ref:`JAX <quickstart-jax>` | :ref:`Pandas <quickstart-pandas>` | :ref:`fastai <quickstart-fastai>` | :ref:`PyTorch Lightning <quickstart-pytorch-lightning>` | :ref:`MXNet <quickstart-mxnet>` | :ref:`scikit-learn <quickstart-scikitlearn>` | :ref:`XGBoost <quickstart-xgboost>`
```
Comment (danieljanes) [inline doc/source/index.rst:58]: The tutorial needs to be added here as well

File: doc/source/quickstart-ios.rst [hunk 2]
```diff
@@ -0,0 +1,151 @@
+.. _quickstart-ios:
+
+
+Quickstart iOS
+==================
```
Comment (danieljanes) [inline doc/source/quickstart-ios.rst:5]:
```suggestion
==============
```

File: doc/source/quickstart-ios.rst [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+First of all, it is recommended to create a virtual environment and run everything within a `virtualenv <https://flower.dev/docs/recommended-env-setup.html>`_. 
+
+Our example consists of one *server* and *clients* that all have the same model. 
+
+*Clients* are responsible for generating individual weight-updates for the model based on their local datasets. 
+These updates are then sent to the *server* which will aggregate them to produce a better model. Finally, the *server* sends this improved version of the model back to each *client*.
+A complete cycle of weight updates is called a *round*.
+
+Now that we have a rough idea of what is going on, let's get started. We first need to install Flower. You can do this by running :
```
Comment (danieljanes) [inline doc/source/quickstart-ios.rst:17]:
```suggestion
Now that we have a rough idea of what is going on, let's get started. We first need to install Flower. You can do this by running:
```

File: doc/source/quickstart-ios.rst [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+.. code-block:: shell
+
+  $ pip install flwr
+
+Or simply install all dependencies using Poetry:
+
+.. code-block:: shell
+
+  $ poetry install
```
Comment (danieljanes) [inline doc/source/quickstart-ios.rst:27]:
```suggestion
  $ poetry add flwr
```

File: doc/source/index.rst [hunk 5]
```diff
@@ -53,6 +53,7 @@ A learning-oriented series of federated learning tutorials, the best place to st
    quickstart-mxnet
    quickstart-scikitlearn
    quickstart-xgboost
+   quickstart-ios
```
Comment (danieljanes) [inline doc/source/index.rst:56]: Let's also add this to the main README

**Review by danieljanes — CHANGES_REQUESTED** [review]

File: doc/source/quickstart-ios.rst [hunk 6]
```diff
@@ -0,0 +1,153 @@
+.. _quickstart-ios:
+
+
+Quickstart iOS
+==============
+
+In this tutorial we will learn how to train a Neural Network on MNIST using Flower and CoreML in iOS devices. 
+
+First of all, for running the flower Python server, it is recommended to create a virtual environment and run everything within a `virtualenv <https://flower.dev/docs/recommended-env-setup.html>`_.
+For the flower client implementation in iOS, it is recommended to use Xcode as our IDE.
```
Comment (danieljanes) [inline doc/source/quickstart-ios.rst:10]: `Flower` should always be capitalized (i.e., `Flower` instead of `flower`).

Comment (danieljanes) [reply doc/source/quickstart-ios.rst:10]: Please also ensure this is the case in the rest of the doc

File: doc/source/quickstart-ios.rst [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+Our example consists of one Python *server* and iPhone *clients* that all have the same model.
```
Comment (danieljanes) [inline doc/source/quickstart-ios.rst:12]: How many iOS clients?

**Review by danieljanes — CHANGES_REQUESTED** [review]

File: README.md [hunk 8]
```diff
@@ -83,6 +83,7 @@ Stay tuned, more tutorials are coming soon. Topics include **Privacy and Securit
 * [Quickstart (JAX)](https://github.com/adap/flower/tree/main/examples/quickstart_jax)
 * [Quickstart (scikit-learn)](https://github.com/adap/flower/tree/main/examples/sklearn-logreg-mnist)
 * [Quickstart (TFLite on Android [code example])](https://github.com/adap/flower/tree/main/examples/android)
+* [Quickstart (iOS)](https://github.com/adap/flower/tree/main/examples/ios)
```
Comment (danieljanes) [inline README.md:86]: This should link to the docs page, not the code example (we only link to the code example if no docs page is available).

File: doc/source/quickstart-ios.rst [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+==============
+
+In this tutorial we will learn how to train a Neural Network on MNIST using Flower and CoreML in iOS devices. 
+
+First of all, for running the flower Python server, it is recommended to create a virtual environment and run everything within a `virtualenv <https://flower.dev/docs/recommended-env-setup.html>`_.
+For the flower client implementation in iOS, it is recommended to use Xcode as our IDE.
+
+Our example consists of one Python *server* and iPhone *clients* that all have the same model. 
+
+*Clients* are responsible for generating individual weight-updates for the model based on their local datasets.
```
Comment (danieljanes) [inline doc/source/quickstart-ios.rst:14]:
```suggestion
*Clients* are responsible for generating individual weight updates for the model based on their local datasets. 
```

File: doc/source/quickstart-ios.rst (hunk 4 above)
Comment (danielnugraha) [reply doc/source/quickstart-ios.rst]: Got the point, changed it back to poetry add flwr and pip install flwr.

File: doc/source/quickstart-ios.rst [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+  ///
+  /// - Returns: The result from the evaluation, e.g., loss
+  public func evaluate(ins: EvaluateIns) -> EvaluateRes {
+    let status = Status(code: .ok, message: String())
+    let result = runMLTask(configuration: parameters.parametersToWeights(parameters: ins.parameters), task: .test)
+        
+    return EvaluateRes(loss: Float(result.loss), numExamples: result.numSamples, status: status)
+  }
+
+Let's create a new application project in Xcode and add flwr as a dependency in your project. For our application, we will store the logic of our app in :code:`FLiOSModel.swift` and the UI elements in :code:`ContentView.swift`.
```
Comment (danieljanes) [inline doc/source/quickstart-ios.rst:69]:
```suggestion
Let's create a new application project in Xcode and add ``flwr`` as a dependency in your project. For our application, we will store the logic of our app in :code:`FLiOSModel.swift` and the UI elements in :code:`ContentView.swift`.
```

File: doc/source/quickstart-ios.rst [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+  // inspect the model to be able to access the model parameters
+  // to access the model we need to know the layer name
+  // since the model parameters are stored as key value pairs
+  let modelInspect = try MLModelInspect(serializedData: Data(contentsOf: url))
+  let layerWrappers = modelInspect.getLayerWrappers()
+  self.mlFlwrClient = MLFlwrClient(layerWrappers: layerWrappers,
+                                   dataLoader: dataLoader,
+                                   compiledModelUrl: compiledModelUrl)
+
+Then start the Flower grpc client and start communicating to the server by passing our Flower client to the function :code:`startFlwrGRPC`.
```
Comment (danieljanes) [inline doc/source/quickstart-ios.rst:114]:
```suggestion
Then start the Flower gRPC client and start communicating to the server by passing our Flower client to the function :code:`startFlwrGRPC`.
```

**Review by danieljanes — APPROVED** [review]
Lgtm!

**Pull request merged**